Improve the handling of oscillating sequences in limit_seq

normalhuman opened the pull request:
#### References to other Issues or PRs

Closes #14198 by including those commits from a now-deleted repository. Addresses some (not all) of the issues with oscillatory sequences in #14192 

#### Brief description of what is fixed or changed

Oscillatory sequences are handled by using odd and even dummy indices (this is more efficient than replacing n by `2*n` or `2*n+1` as was previously done). Also, the presence of sin or cos in the formula now also qualifies a sequence as oscillatory, in addition to powers with a negative base.

When the limits along even and odd indices exist but are different, AccumBounds is returned instead of None.

For some sequences, the absolute value can be handled by limit_seq while the original sequence cannot (for example, (I/2)**n). If the limit of absolute value is 0, the limit of the original sequence is also
0.

For example, `limit_seq((-S.Half)**n)` returned nothing previously, now it returns 0.

#### Release Notes

<!-- BEGIN RELEASE NOTES -->
* series
   * Improved the computation of limits of oscillating sequences 
<!-- END RELEASE NOTES -->

normalhuman committed:
Improve the handling of oscillatory sequences in limit_seq

Oscillatory sequences are handled by considering odd and even indices
separately (this is more efficient than replacing n by `2*n` or `2*n+1`
as was previously done). Also, the presence of sin or cos in the formula
now qualifies a sequence as oscillatory.

When the limits along even and odd indices exist but are different,
AccumBounds is returned instead of None.

For some sequences, the absolute value can be handled by `limit_seq`
while the original sequence cannot (for example, `(I/2)**n`). If the
limit of absolute value is 0, the limit of the original sequence is also
0.

jksuom commented on sympy/series/limitseq.py:
```diff
@@ -193,29 +196,45 @@ def limit_seq(expr, n=None, trials=5):
-                return L1
+    if (any(b.is_negative and e.has(n) for b, e in powers) or
+        expr.has(cos, sin)):
+            L1 = _limit_seq(expr.xreplace({n: n1}), n1, trials)
```
It would probably suffice to indent 4 spaces wrt `if ...`.

jksuom commented on sympy/series/limitseq.py:
```diff
@@ -193,29 +196,45 @@ def limit_seq(expr, n=None, trials=5):
-        return _limit_seq(expr.xreplace({n: n_}), n_, trials)
+        # Maybe the absolute value is easier to deal with (though not if
+        # it's a sum). If it tends to 0, the limit is 0.
+        if not(expr.has(Sum) or expr.is_Add):
```
Could sums like `n/(n + 1) + (I/2)**n` be handled by looking at each term separately?

normalhuman committed:
Split an Add if initially unsuccessful

Some expressions, like `n/(n+1) + (-1/2)**n`, have oscillating and
non-oscillating parts, which require different approaches. In this
example evaluation succeeds by taking Abs of the second argument only.

jksuom commented:
Thanks, looks good.
